python/cugraph/cugraph/link_prediction/overlap.py

```diff
@@ -11,28 +11,96 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from cugraph.link_prediction import overlap_wrapper
-import cudf
 from cugraph.utilities import (
     ensure_cugraph_obj_for_nx,
     df_edge_score_to_dictionary,
     renumber_vertex_pair,
 )
+import cudf
+import warnings
+
+from pylibcugraph import (
+    overlap_coefficients as pylibcugraph_overlap_coefficients,
+)
+from pylibcugraph import ResourceHandle
 
 
-def overlap_coefficient(G, ebunch=None, do_expensive_check=True):
+# FIXME: Move this function to the utility module so that it can be
+# shared by other algos
+def ensure_valid_dtype(input_graph, vertex_pair):
+    vertex_dtype = input_graph.edgelist.edgelist_df.dtypes[0]
+    vertex_pair_dtypes = vertex_pair.dtypes
+
+    if vertex_pair_dtypes[0] != vertex_dtype or vertex_pair_dtypes[1] != vertex_dtype:
+        warning_msg = (
+            "Overlap requires 'vertex_pair' to match the graph's 'vertex' type. "
+            f"input graph's vertex type is: {vertex_dtype} and got "
+            f"'vertex_pair' of type: {vertex_pair_dtypes}."
+        )
+        warnings.warn(warning_msg, UserWarning)
+        vertex_pair = vertex_pair.astype(vertex_dtype)
+
+    return vertex_pair
+
+
+def overlap_coefficient(G, ebunch=None, do_expensive_check=False):
     """
     For NetworkX Compatability.  See `overlap`
```

Comment:
There is no NetworkX API for `overlap_coefficient` so I am wondering if it is worth supporting this API or leaving it as it is with the expectation that NetworkX will later provide one matching it?